src/screenshot.cpp

```diff
@@ -668,43 +668,40 @@ static void LargeWorldCallback(void *userdata, void *buf, uint y, uint pitch, ui
  */
 static const char *MakeScreenshotName(const char *default_fn, const char *ext, bool crashlog = false)
 {
-	bool generate = StrEmpty(_screenshot_name);
+	bool generate = _screenshot_name.empty();
 
 	if (generate) {
 		if (_game_mode == GM_EDITOR || _game_mode == GM_MENU || _local_company == COMPANY_SPECTATOR) {
-			strecpy(_screenshot_name, default_fn, lastof(_screenshot_name));
+			_screenshot_name = default_fn;
 		} else {
-			strecpy(_screenshot_name, GenerateDefaultSaveName().c_str(), lastof(_screenshot_name));
+			_screenshot_name = GenerateDefaultSaveName();
 		}
 	}
 
-	size_t len = strlen(_screenshot_name);
-
 	/* Handle user-specified filenames ending in # with automatic numbering */
 	if (StrEndsWith(_screenshot_name, "#")) {
 		generate = true;
-		len -= 1;
-		_screenshot_name[len] = '\0';
+		_screenshot_name.pop_back();
 	}
 
+	size_t len = _screenshot_name.size();
 	/* Add extension to screenshot file */
-	seprintf(&_screenshot_name[len], lastof(_screenshot_name), ".%s", ext);
+	_screenshot_name += fmt::format(".{}", ext);
 
 	const char *screenshot_dir = crashlog ? _personal_dir.c_str() : FiosGetScreenshotDir();
 
 	for (uint serial = 1;; serial++) {
-		if (seprintf(_full_screenshot_name, lastof(_full_screenshot_name), "%s%s", screenshot_dir, _screenshot_name) >= (int)lengthof(_full_screenshot_name)) {
-			/* We need more characters than MAX_PATH -> end with error */
-			_full_screenshot_name[0] = '\0';
-			break;
-		}
+		_full_screenshot_path = screenshot_dir;
+		_full_screenshot_path += _screenshot_name;
```

Comment:
Just curious: `fmt::format("{}{}", screenshot_dir, _screenshot_name)` is a bit more readable, I think. But I can imagine that the current format is faster? Is there a reason for one over the other?

Reply:
I'm not sure which one's faster, as that might highly depend on the situation. Though screenshot file names are not a place where we should really focus on the performance, but rather readability and functionality.